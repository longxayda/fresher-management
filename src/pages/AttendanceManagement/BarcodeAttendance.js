import React from "react";
import {QRCodeSVG} from 'qrcode.react';
import setTime from "components/Attendance/changedUrl";
import { Container } from "reactstrap";

function BarcodeAttendance() {
  let unchangeUrl = "https://fresher-management.herokuapp.com/"
  let changedUrl = "attend"
  let url = unchangeUrl + changedUrl

  return (
    <>
      <Container>
          <div style={{textAlign: "center"}}>
          <h4 style={{margin: "0 0 30px 0"}}>Please scan this QR code to fill in your attend form</h4>
          <QRCodeSVG
            style = {{height: "300px", width: "300px", display: "flex", margin: "auto"}}
            value = {url}
          />
        </div>
      </Container>
    </>
  );
}

export default BarcodeAttendance;