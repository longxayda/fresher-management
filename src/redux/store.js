import {configureStore} from "@reduxjs/toolkit";
import attendTableOneTraineeSlice from "./attendanceManagementSlice/attendTableOneTraineeSlice";

// import ticketClassSlice from "./attendanceManagementSlice/ticketClassSlice";
import diligentSummarySlice from "./attendanceManagementSlice/diligentSummarySlice";
import leaveRequestSlice from "./attendanceManagementSlice/leaveRequestSlice";
import checkboxFRReducer from "./fresherManageSlice/checkboxFresherSlice";
import evaluationSlice from "./auditorsSlice/evaluationSlice";
import resourceSlice from "./auditorsSlice/resourceSlice";
import auditorSlice from "./auditorsSlice/auditorSlice";
import manageFRReducer from "./fresherManageSlice/fresherManagementSlice";
import bonusPenaltyReducer from "./fresherManageSlice/penaltySlice";

import classSlice from "./classManagementSlice/classSlice";
import classDetailsSlice from "./classManagementSlice/classDetailsSlice";
import learningPathSlice from "./classManagementSlice/learningPathSlice";
import syllabusSlice from "./classManagementSlice/syllabusSlice";
import moduleSlice from "./classManagementSlice/moduleSlice";
import topicSlice from "./classManagementSlice/topicSlice";

import scoreSlice from "./fresherManageSlice/scoreSlice";
import topicMarkSlice from "./fresherManageSlice/topicMarkSlice";
import scheduleSlice from "./auditorsSlice/scheduleSlice";
import createNewAuditSlice from "./auditorsSlice/createNewAuditSlice";
import adminManageAttendanceReducer from "./attendanceManagementSlice/adminManageAttendanceSlice";
import adminTicketReducer from "./attendanceManagementSlice/adminTicketSlice";

import traineeAttendanceTableSlice from "./attendanceManagementSlice/traineeAttendanceTableSlice";
import authSlice from "./authSlice/authSlice";
import roleSlice from "./roleSlice";
import UserListSlice from "./UserListSlices/UserListSlice";
import attendanceSlice from "./attendanceManagementSlice/attendanceSlice";
import traineeUserProfileSlice from "./userProfileSlice/traineeUserProfileSlice";
import adminUserProfileSlice from "./userProfileSlice/adminUserProfileSlice";

import AddNewReportTableSlice from "components/WeeklyReportTables/addNewReportTableSlice";
import DetailReportTableSlice from "components/WeeklyReportTables/detailReportTableSlice";
import ReportListTableSlice from "components/WeeklyReportTables/reportListTableSlice";
import searchSlice from "components/Search/searchSlice";
import ReportModalSlice from "components/ReportModal/reportModalSlice";

import HistoryReportListTableSlice from "components/MonthlyReportTables/historyReportListTableSlice";
import coursesReportSlice from "../pages/MonthlyReportPage/MonthlyHistoryReport/courses/coursesReportSlice";
import financeReportSlice from "../pages/MonthlyReportPage/MonthlyHistoryReport/finance/financeReportSlice";
import detailCoursesReportSlice from "../pages/MonthlyReportPage/MonthlyHistoryReport/courses/detailCourseReportSlice";
import detailFinanceReportSlice from "../pages/MonthlyReportPage/MonthlyHistoryReport/finance/detailFinanceReportSlice";

import DetailDeliveryTableSlice from "components/MonthlyReportTables/detailDeliveryTableSlice";
import DeliveryReportListTableSlice from "components/MonthlyReportTables/deliveryReportListTableSlice";
import NewDeliveryTableSlice from "components/MonthlyReportTables/newDeliveryTableSlice";

import OutputsReportTableSlice from "components/OutputsReportTables/outputsReportTableSlice";
import fresherReportSlice from "../pages/OutputsReportPage/fresher/fresherReportSlice";
import svttReportSlice from "../pages/OutputsReportPage/svtt/svttReportSlice";

import detailOutputsReportSlice from "../pages/OutputsReportPage/detailOutputsReportSlice";
import detailFresherReportSlice from "../pages/OutputsReportPage/fresher/detailFresherReportSlice";
import detailSVTTReportSlice from "../pages/OutputsReportPage/svtt/detailSVTTReportSlice";

import thunk from "redux-thunk";
const store = configureStore({
	reducer: {
		Filters: searchSlice,
		evaluation: evaluationSlice,
		resource: resourceSlice,
		manageFR: manageFRReducer,
		checkboxFR: checkboxFRReducer,
		attendTableOneTrainee: attendTableOneTraineeSlice,
		// traineeTabTable: TraineeTabTableSlice,
		class: classSlice,
		learningPath: learningPathSlice,
		syllabus: syllabusSlice,
		classDetails: classDetailsSlice,
		module: moduleSlice,
		topicModule: topicSlice,
		auth: authSlice,
		role: roleSlice,
		// UserList
		UserList: UserListSlice,
		schedule: scheduleSlice,
		newAudit: createNewAuditSlice,
		auditor: auditorSlice,

		// attendTable: attendTableSlice,
		// ticketClass: ticketClassSlice,
		diligentSummary: diligentSummarySlice,
		leaveRequest: leaveRequestSlice,
		adminManageAttendance: adminManageAttendanceReducer,
		adminTicket: adminTicketReducer,
		attendance: attendanceSlice,
		traineeAttendanceTable: traineeAttendanceTableSlice,
		bonusPenalty: bonusPenaltyReducer,
		score: scoreSlice,
		topic: topicMarkSlice,
		traineeUserProfile: traineeUserProfileSlice,
		adminUserProfile: adminUserProfileSlice,

		ReportList: ReportListTableSlice,
		AddNewReport: AddNewReportTableSlice,
		DetailReport: DetailReportTableSlice,
		Filters: searchSlice,
		ReportModal: ReportModalSlice,

		HistoryReportList: HistoryReportListTableSlice,
		CoursesReport: coursesReportSlice,
		FinanceReport: financeReportSlice,
		DetailCoursesReport: detailCoursesReportSlice,
		DetailFinanceReport: detailFinanceReportSlice,
		DetailDeliveryReport: DetailDeliveryTableSlice,
		DeliveryReportList: DeliveryReportListTableSlice,
		NewDeliveryReport: NewDeliveryTableSlice,

		OutputsReport: OutputsReportTableSlice,
		FresherReport: fresherReportSlice,
		SVTTReport: svttReportSlice,
		DetailFresherReport: detailFresherReportSlice,
		DetailSVTTReport: detailSVTTReportSlice,
		DetailOutputsReport: detailOutputsReportSlice,
	},
	middleware: [thunk],
});

export default store;
