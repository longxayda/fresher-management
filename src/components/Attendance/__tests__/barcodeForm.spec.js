import React from 'react';
import Enzyme from 'enzyme'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { mount } from 'enzyme';
Enzyme.configure({ adapter: new Adapter() });
import store from 'redux/store';
import { Provider } from "react-redux";
import BarcodeAttendance from 'pages/AttendanceManagement/BarcodeAttendance.js'

describe('Test Attendance Form Sreen', () => {
    let component;
    beforeEach(() => {
        component = mount(<Provider store = { store }> <BarcodeAttendance /> </Provider>);
    });

    test('has the correct class', () => {
        expect(component.find('.barcode-form').exists()).toEqual(true);
    });

    test('render Barcode Form Sreen\'s component correctly', () => {
        component = mount(<Provider store = { store }> <BarcodeAttendance /> </Provider>);
        expect(component.html()).toMatchSnapshot();
    });
})
