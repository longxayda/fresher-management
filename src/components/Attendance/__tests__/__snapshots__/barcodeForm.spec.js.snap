// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Attendance Form Sreen render Barcode Form Sreen's component correctly 1`] = `" <div class=\\"barcode-form\\" style=\\"text-align: center;\\"><h4 style=\\"margin: 0px 0px 30px 0px;\\">Please scan this QR code to fill in your attend form</h4><svg height=\\"128\\" width=\\"128\\" viewBox=\\"0 0 29 29\\" style=\\"height: 300px; width: 300px; display: flex; margin: auto;\\"><path fill=\\"#FFFFFF\\" d=\\"M0,0 h29v29H0z\\" shape-rendering=\\"crispEdges\\"></path><path fill=\\"#000000\\" d=\\"M0 0h7v1H0zM10 0h1v1H10zM13 0h7v1H13zM22,0 h7v1H22zM0 1h1v1H0zM6 1h1v1H6zM8 1h3v1H8zM13 1h6v1H13zM20 1h1v1H20zM22 1h1v1H22zM28,1 h1v1H28zM0 2h1v1H0zM2 2h3v1H2zM6 2h1v1H6zM8 2h1v1H8zM16 2h2v1H16zM20 2h1v1H20zM22 2h1v1H22zM24 2h3v1H24zM28,2 h1v1H28zM0 3h1v1H0zM2 3h3v1H2zM6 3h1v1H6zM9 3h1v1H9zM12 3h1v1H12zM14 3h3v1H14zM18 3h2v1H18zM22 3h1v1H22zM24 3h3v1H24zM28,3 h1v1H28zM0 4h1v1H0zM2 4h3v1H2zM6 4h1v1H6zM8 4h3v1H8zM12 4h5v1H12zM18 4h1v1H18zM22 4h1v1H22zM24 4h3v1H24zM28,4 h1v1H28zM0 5h1v1H0zM6 5h1v1H6zM8 5h1v1H8zM10 5h1v1H10zM12 5h1v1H12zM14 5h2v1H14zM18 5h2v1H18zM22 5h1v1H22zM28,5 h1v1H28zM0 6h7v1H0zM8 6h1v1H8zM10 6h1v1H10zM12 6h1v1H12zM14 6h1v1H14zM16 6h1v1H16zM18 6h1v1H18zM20 6h1v1H20zM22,6 h7v1H22zM8 7h1v1H8zM10 7h2v1H10zM14 7h7v1H14zM0 8h2v1H0zM3 8h1v1H3zM6 8h2v1H6zM11 8h4v1H11zM22 8h3v1H22zM26 8h2v1H26zM0 9h1v1H0zM2 9h3v1H2zM7 9h1v1H7zM9 9h1v1H9zM11 9h2v1H11zM16 9h1v1H16zM20 9h3v1H20zM25 9h1v1H25zM28,9 h1v1H28zM0 10h2v1H0zM3 10h1v1H3zM5 10h4v1H5zM11 10h2v1H11zM16 10h2v1H16zM19 10h1v1H19zM22 10h2v1H22zM25 10h3v1H25zM0 11h2v1H0zM3 11h2v1H3zM7 11h1v1H7zM9 11h5v1H9zM15 11h1v1H15zM19 11h3v1H19zM26 11h2v1H26zM1 12h4v1H1zM6 12h1v1H6zM10 12h4v1H10zM16 12h2v1H16zM20 12h4v1H20zM25 12h1v1H25zM27,12 h2v1H27zM0 13h1v1H0zM2 13h1v1H2zM4 13h2v1H4zM9 13h1v1H9zM11 13h1v1H11zM13 13h1v1H13zM17 13h5v1H17zM0 14h8v1H0zM9 14h1v1H9zM11 14h2v1H11zM14 14h1v1H14zM17 14h1v1H17zM21 14h2v1H21zM24,14 h5v1H24zM0 15h5v1H0zM8 15h1v1H8zM10 15h8v1H10zM20 15h1v1H20zM25 15h1v1H25zM27 15h1v1H27zM0 16h2v1H0zM3 16h1v1H3zM6 16h2v1H6zM9 16h1v1H9zM12 16h1v1H12zM14 16h1v1H14zM16 16h3v1H16zM20 16h2v1H20zM27 16h1v1H27zM1 17h1v1H1zM3 17h1v1H3zM8 17h2v1H8zM11 17h2v1H11zM14 17h1v1H14zM16 17h2v1H16zM22 17h2v1H22zM25 17h1v1H25zM28,17 h1v1H28zM0 18h1v1H0zM3 18h1v1H3zM5 18h2v1H5zM8 18h1v1H8zM12 18h2v1H12zM17 18h1v1H17zM22 18h1v1H22zM27,18 h2v1H27zM2 19h1v1H2zM4 19h2v1H4zM7 19h2v1H7zM10 19h1v1H10zM12 19h1v1H12zM14 19h1v1H14zM16 19h2v1H16zM21 19h1v1H21zM23 19h1v1H23zM27,19 h2v1H27zM0 20h1v1H0zM6 20h1v1H6zM10 20h1v1H10zM12 20h5v1H12zM20 20h5v1H20zM26 20h1v1H26zM8 21h1v1H8zM10 21h1v1H10zM12 21h2v1H12zM16 21h1v1H16zM19 21h2v1H19zM24 21h1v1H24zM26,21 h3v1H26zM0 22h7v1H0zM8 22h1v1H8zM13 22h2v1H13zM16 22h1v1H16zM18 22h1v1H18zM20 22h1v1H20zM22 22h1v1H22zM24 22h1v1H24zM27 22h1v1H27zM0 23h1v1H0zM6 23h1v1H6zM9 23h2v1H9zM12 23h2v1H12zM16 23h1v1H16zM20 23h1v1H20zM24 23h3v1H24zM28,23 h1v1H28zM0 24h1v1H0zM2 24h3v1H2zM6 24h1v1H6zM10 24h1v1H10zM12 24h2v1H12zM15 24h3v1H15zM20 24h5v1H20zM27 24h1v1H27zM0 25h1v1H0zM2 25h3v1H2zM6 25h1v1H6zM8 25h2v1H8zM12 25h1v1H12zM14 25h1v1H14zM17 25h1v1H17zM19 25h1v1H19zM21 25h2v1H21zM24 25h4v1H24zM0 26h1v1H0zM2 26h3v1H2zM6 26h1v1H6zM9 26h1v1H9zM11 26h2v1H11zM14 26h1v1H14zM16 26h2v1H16zM19 26h2v1H19zM24 26h3v1H24zM28,26 h1v1H28zM0 27h1v1H0zM6 27h1v1H6zM8 27h1v1H8zM10 27h4v1H10zM17 27h4v1H17zM22 27h3v1H22zM27 27h1v1H27zM0 28h7v1H0zM8 28h2v1H8zM16 28h2v1H16zM20 28h2v1H20zM23 28h3v1H23zM27 28h1v1H27z\\" shape-rendering=\\"crispEdges\\"></path></svg></div> "`;
